### ISSUE TITLE
Problem with creating new flutter projects in VS code

### ISSUE THREAD

**Issue by efidje** [flutter/flutter#16926]
## Steps to Reproduce

Please tell us what you were doing and what went wrong. If you are running flutter tools from the command line, please try adding the `-v` or `--verbose` option to gather more information.

**_I was attempting to create a new flutter project in VS code but just got this message:_
Oops; flutter has exited _unexpectedly._
Sending crash report to Google.**_

If the problem is with your application's rendering, please attach a screenshot and any relevant source code.
If you are getting an exception in the logs, and your code is implicated in the first few frames, then please include the source code for the functions involved.

## Logs

Run your application with `flutter run` and attach all the log output.

C:\Users\Endre\App-projects\Flutter_test\my_flutter_test>flutter run
Running "flutter packages get" in my_flutter_test...

Oops; flutter has exited unexpectedly.
Sending crash report to Google.
Crash report sent (report ID: 582d862e9e93533d)
Crash report written to C:\Users\Endre\App-projects\Flutter_test\my_flutter_test\flutter_01.log;
please let us know at https://github.com/flutter/flutter/issues.

Run `flutter analyze` and attach any output of that command also.
C:\Users\Endre\App-projects\Flutter_test\my_flutter_test>flutter analyze
Running "flutter packages get" in my_flutter_test...
Oops; flutter has exited unexpectedly.
Sending crash report to Google.
Crash report sent (report ID: 658c2cf688134868)
Crash report written to C:\Users\Endre\App-projects\Flutter_test\my_flutter_test\flutter_02.log;

## Flutter Doctor

Paste the output of running `flutter doctor -v` here.
C:\Users\Endre\App-projects\Flutter_test\my_flutter_test> flutter doctor -v
[√] Flutter (Channel beta, v0.2.8, on Microsoft Windows [Version 6.1.7601], locale nb-NO)
    • Flutter version 0.2.8 at C:\Users\Endre\App-projects\flutter
    • Framework revision b397406561 (3 weeks ago), 2018-04-02 13:53:20 -0700
    • Engine revision c903c217a1
    • Dart version 2.0.0-dev.43.0.flutter-52afcba357

[√] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at C:\Users\Endre\AppData\Local\Android\sdk
    • Android NDK at C:\Users\Endre\AppData\Local\Android\sdk\ndk-bundle
    • Platform android-27, build-tools 27.0.3
    • ANDROID_HOME = C:\Users\Endre\AppData\Local\Android
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b02)
    • All Android licenses accepted.

[√] Android Studio (version 3.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b02)

[√] VS Code, 64-bit edition (version 1.22.2)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Dart Code extension version 2.11.2

[√] Connected devices (1 available)
    • A701 • 03501aa9785500000000 • android-arm • Android 4.4.2 (API 19)

• No issues found!

> For more information about diagnosing and reporting Flutter bugs, please see [https://flutter.io/bug-reports/](https://flutter.io/bug-reports/).
[flutter_01.log](https://github.com/flutter/flutter/files/1942608/flutter_01.log)

**Comment by gloryluu** [comment]
I'm using the master branch without any issue. Why don't you try using the latest version 0.33.

**Comment by DanTup** [comment]
This doesn't seem VS Code related, seems to be crashing from the CLI; attached log starts with:

## command

flutter create my_flutter_test

## exception

ArgumentError: Invalid argument(s): Cannot find executable for C:\Users\Endre\App-projects\flutter\bin\cache\dart-sdk\bin\pub.bat.

```
#0      _getExecutable (package:process/src/interface/local_process_manager.dart:113)
#1      LocalProcessManager.start (package:process/src/interface/local_process_manager.dart:41)
#2      runCommand (package:flutter_tools/src/base/process.dart:115)
#3      runCommandAndStreamOutput (package:flutter_tools/src/base/process.dart:133)
<asynchronous suspension>
```

@efidje Do you have a file at `C:\Users\Endre\App-projects\flutter\bin\cache\dart-sdk\bin\pub.bat` on your disk?

**Comment by efidje** [comment]
Gents,

Thank you for the feedback. Not sure what happened. The "create new flutter project" was started from inside VS code. This option had worked previously without any issues. I checked the C:\Users\Endre\App-projects\flutter\bin\cache\dart-sdk\bin\ location as suggested by DanTup and found that indeed there was no pub.bat file in that location. I proceeded to download the flutter package again from the internet on did an overwrite of the old installation. That worked and I now can create flutter projects from inside VS code again. Can't see what caused the pub.bat file to go missing in the first place, strange...
